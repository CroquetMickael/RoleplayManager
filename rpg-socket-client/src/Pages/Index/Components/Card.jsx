import React from "react";
import { useNavigate } from "react-router";

const CardGame = (props) => {
  const navigation = useNavigate();
  return (
    <div className="w-1/3 bg-white rounded-lg shadow-md ">
      <div className="p-4">
        <div className="flex items-center justify-between">
          <h2 className="-mt-1 text-lg font-semibold text-gray-900">
            {props.roomName}
          </h2>
          <small className="text-sm text-gray-700">
            {props.playersNumber}/{props.roomMaxPlayer}
          </small>
        </div>
        <p className="pt-3 text-sm text-gray-700">
          players :{" "}
          {props.players.map((player) => (
            <span key={player.name}>{player.name} </span>
          ))}
        </p>
        <button
          className="w-full py-2 my-2 text-white bg-blue-400 rounded-sm hover:bg-blue-600"
          onClick={() => {
            navigation(`/game/${props.roomName}`);
          }}
        >
          {" "}
          Join game
        </button>
      </div>
    </div>
  );
};

export { CardGame };
