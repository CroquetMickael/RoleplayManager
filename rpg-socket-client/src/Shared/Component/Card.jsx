import React from "react";

const Card = ({ children, leftSidetext, rightSideText, isVertical }) => (
  <div className="mx-4 my-2 bg-white rounded-lg shadow-md">
    <div className="p-4">
      <div
        className={`flex items-center justify-between  border-black ${
          isVertical ? "flex-col" : "border-b"
        }`}
      >
        <h2 className="-mt-1 text-lg font-semibold text-gray-900">
          {leftSidetext}
        </h2>
        <small className="text-sm text-gray-700">{rightSideText}</small>
      </div>
    </div>
    {children}
  </div>
);

export { Card };
