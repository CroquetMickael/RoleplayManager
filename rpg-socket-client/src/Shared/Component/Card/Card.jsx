import React from "react";

const Card = ({ children, leftSidetext, rightSideText, isVertical }) => (
  <div className="mx-4 my-2 text-black bg-white rounded-lg shadow-md dark:bg-gray-700 dark:text-white">
    <div className="p-4">
      <div
        className={`flex items-center justify-between  border-black ${
          isVertical ? "flex-col" : "border-b"
        }`}
      >
        <h2 className="-mt-1 text-lg font-semibold text-gray-900 dark:text-white">
          {leftSidetext}
        </h2>
        <small className="text-sm text-gray-900 dark:text-white">{rightSideText}</small>
      </div>
    </div>
    {children}
  </div>
);

export { Card };
