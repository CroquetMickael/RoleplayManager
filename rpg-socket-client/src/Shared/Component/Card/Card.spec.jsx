import React from "react";
import { render } from "@testing-library/react";
import { Card } from "./Card";

it("renders correctly a vertical Card with left, right text", () => {
  const tree = render(
    <Card
      isVertical={true}
      leftSidetext={"lefttext"}
      rightSideText={"righttext"}
    >
      children
    </Card>
  ).container;
  expect(tree).toMatchSnapshot();
});

it("renders correctly a vertical Card without left, right text", () => {
  const tree = render(<Card isVertical={true}>children</Card>).container;
  expect(tree).toMatchSnapshot();
});

it("renders correctly a non vertical Card with left, right text", () => {
  const tree = render(
    <Card
      isVertical={false}
      leftSidetext={"lefttext"}
      rightSideText={"righttext"}
    >
      children
    </Card>
  ).container;
  expect(tree).toMatchSnapshot();
});

it("renders correctly a non vertical Card without left, right text", () => {
  const tree = render(<Card isVertical={false}>children</Card>).container;
  expect(tree).toMatchSnapshot();
});
