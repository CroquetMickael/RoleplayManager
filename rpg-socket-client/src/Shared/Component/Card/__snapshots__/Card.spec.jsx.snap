// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly a non vertical Card with left, right text 1`] = `
<div>
  <div
    class="mx-4 my-2 text-black bg-white rounded-lg shadow-md dark:bg-gray-700 dark:text-white"
  >
    <div
      class="p-4"
    >
      <div
        class="flex items-center justify-between  border-black border-b"
      >
        <h2
          class="-mt-1 text-lg font-semibold text-gray-900 dark:text-white"
        >
          lefttext
        </h2>
        <small
          class="text-sm text-gray-900 dark:text-white"
        >
          righttext
        </small>
      </div>
    </div>
    children
  </div>
</div>
`;

exports[`renders correctly a non vertical Card without left, right text 1`] = `
<div>
  <div
    class="mx-4 my-2 text-black bg-white rounded-lg shadow-md dark:bg-gray-700 dark:text-white"
  >
    <div
      class="p-4"
    >
      <div
        class="flex items-center justify-between  border-black border-b"
      >
        <h2
          class="-mt-1 text-lg font-semibold text-gray-900 dark:text-white"
        />
        <small
          class="text-sm text-gray-900 dark:text-white"
        />
      </div>
    </div>
    children
  </div>
</div>
`;

exports[`renders correctly a vertical Card with left, right text 1`] = `
<div>
  <div
    class="mx-4 my-2 text-black bg-white rounded-lg shadow-md dark:bg-gray-700 dark:text-white"
  >
    <div
      class="p-4"
    >
      <div
        class="flex items-center justify-between  border-black flex-col"
      >
        <h2
          class="-mt-1 text-lg font-semibold text-gray-900 dark:text-white"
        >
          lefttext
        </h2>
        <small
          class="text-sm text-gray-900 dark:text-white"
        >
          righttext
        </small>
      </div>
    </div>
    children
  </div>
</div>
`;

exports[`renders correctly a vertical Card without left, right text 1`] = `
<div>
  <div
    class="mx-4 my-2 text-black bg-white rounded-lg shadow-md dark:bg-gray-700 dark:text-white"
  >
    <div
      class="p-4"
    >
      <div
        class="flex items-center justify-between  border-black flex-col"
      >
        <h2
          class="-mt-1 text-lg font-semibold text-gray-900 dark:text-white"
        />
        <small
          class="text-sm text-gray-900 dark:text-white"
        />
      </div>
    </div>
    children
  </div>
</div>
`;
